### PR TITLE
feat(cli): support capacitor portals plugin changes needed

### PR DESCRIPTION
**Pull request by carlpoole**

### CODE REVIEW

File: cli/src/tasks/copy.ts [hunk 1]
```diff
@@ -70,13 +72,38 @@ export async function copy(
...
+          config.app.webDirAbs,
+        );
+      }
       await copyCapacitorConfig(config, config.ios.nativeTargetDirAbs);
       const cordovaPlugins = await getCordovaPlugins(config, platformName);
       await handleCordovaPluginsJS(cordovaPlugins, config, platformName);
     } else if (platformName === config.android.name) {
-      await copyWebDir(config, config.android.webDirAbs);
+      if (isFederated) {
+        await copyFederatedWebDirs(config, await config.android.webDirAbs);
```
Comment (Steven0351) [inline cli/src/tasks/copy.ts:99]: I noticed here that we're awaiting here and not on line 103, did `config.android.webDirAbs` become async?

Comment (carlpoole) [reply cli/src/tasks/copy.ts]: Nice catch, thanks

**Review by Steven0351 — APPROVED** [review]

File: cli/src/tasks/copy.ts [hunk 2]
```diff
@@ -70,23 +72,55 @@ export async function copy(
       'capacitor:copy:before',
     );
 
+    const allPlugins = await getPlugins(config, platformName);
+    let usesCapacitorPortals = false;
+    if (
+      allPlugins.filter(plugin => plugin.id === '@capacitor/federation')
```
Comment (carlpoole) [inline cli/src/tasks/copy.ts:78]: This needs to be changed to reflect the actual package name we use for Capacitor Portals Plugin

**Review by IT-MikeS — APPROVED** [review]
This looks good to me, nice and clean 👍

**Comment by jthoms1** [comment]
@jcesarmobile Brought up the point that we might use hooks from plugins rather than having the copy code in CLI core. We discussed this as a group and feel like that might cause more issues moving forward.

Hooks for platforms makes more sense because the code mods are only controlled by a single codebase. Electron/iOS/Android.

But in the case of allowing for code mods or hooks from Plugins we run into issues. These hooks can introduce unpredictable side effects. Where the order of plugin loading can yield different results and individual plugins can cause major issues with projects without devs realizing this in the usage.

For this case we believe it is best to continue with this simplistic approach of having the code mods in the CLI core codebase.

**Review by IT-MikeS — APPROVED** [review]

**Review by Steven0351 — APPROVED** [review]

**Pull request merged**